includes/class-webmention-request.php

```diff
@@ -183,10 +183,11 @@ protected function check_content_type( $content_type ) {
 	 * Check if response is a supported content type
 	 *
 	 * @param int $code Status Code.
+	 * @param array $response Response if Present
 	 *
 	 * @return WP_Error|true return an error or that something is supported
 	 */
-	protected function check_response_code( $code ) {
+	protected function check_response_code( $code, $response = null ) {
```

Comment:
If `$response` is optional (default `null`) we should also make the output optional (https://github.com/pfefferle/wordpress-webmention/pull/305/files#diff-ee23ad414e45c72a6f2a72d83420c67460c433f4fa2168ee616a5644047930b3R230). Otherwise wie have to require `$response`.

Reply:
Same for https://github.com/pfefferle/wordpress-webmention/pull/305/files#diff-ee23ad414e45c72a6f2a72d83420c67460c433f4fa2168ee616a5644047930b3R208

Reply:
Otherwise it has to be `array|WP_Error` as default: https://developer.wordpress.org/reference/functions/wp_remote_retrieve_header/

Reply:
> If `$response` is optional (default `null`) we should also make the output optional (https://github.com/pfefferle/wordpress-webmention/pull/305/files#diff-ee23ad414e45c72a6f2a72d83420c67460c433f4fa2168ee616a5644047930b3R230). Otherwise wie have to require `$response`.

Well, the output is always mandatory. In a case where response is null, wp_remote_retrieve_header returns an empty string, so it would still return the WP_Error object, just without the extra information.